### PR TITLE
[#17] Consider using 'clock' package instead of 'getCPUTime'

### PR DESCRIPTION
**Pull request by lucazulian**
Resolves [#17]

### CODE REVIEW

File: three-layer.cabal [hunk 1]
```diff
@@ -135,6 +135,7 @@ library
                      , time >= 1.8 && < 1.10
                      , unordered-containers
                      , warp ^>= 3.2
+                     , clock
```
Comment (chshersh) [inline three-layer.cabal]: Let's also specify lower and upper bounds for this package. Also, it should be in alphabetical order in `build-depends`

File: src/Lib/Effects/Measure.hs [hunk 2]
```diff
@@ -36,10 +36,10 @@ timedActionImpl
        )
     => m a -> m a
 timedActionImpl action = do
-    start <- liftIO getCPUTime
+    start <- liftIO (getTime Monotonic)
     !result <- action
-    end <- liftIO getCPUTime
-    let !timeTaken = fromIntegral (end - start) * 1e-12
+    end <- liftIO (getTime Monotonic)
+    let !timeTaken = fromIntegral (sec end - sec start) * 1e-12
```
Comment (chshersh) [inline src/Lib/Effects/Measure.hs]: Previous `getCPUTime` returned time in picoseconds. That's why it was multiplied by `1e-12`. Now we don't need this multiplication. Instead it's better to conver time to nanoseconds using `toNanoSecs` function and multiply by `1e-9`

**Comment by vrom911** [comment]
Recently I found out that there exist [`GHC.Clock`](http://hackage.haskell.org/package/base-4.12.0.0/docs/GHC-Clock.html) module since base-4.11. Probably we can use its functions without adding any dependencies?

**Comment by chshersh** [comment]
@vrom911 Nice catch! I didn't know about the `GHC.Clock` module. Yes, looks like we don't need to depend on the `clock` package and can just use functions from `base` instead.

**Comment by lucazulian** [comment]
@vrom911 @chshersh awesome, I will use the GHC package!

**Pull request merged**